### ISSUE TITLE
binding make reference to 'this' non consistent with typescript

### ISSUE THREAD

**Issue by jphautin** [aurelia/binding#478]
**I'm submitting a bug report**

* **Library Version:**
aurelia 1.0.0 using aurelia-skeleton-typescript-webpack 1.0.0

**Please tell us about your environment:**
* **Operating System:** 
Linux Debian 
Windows 10
* **Node Version:**
node v6.3.1
npm 3.10.3
* **JSPM OR Webpack AND Version**
webpack 2.1.0-beta.20

* **Browser:**
Iceweasel 38.8.0 | Chrome 52.0

* **Language:**
TypeScript 2.0.0 (also tested with 1.8.10)

**Current behavior:**
I am trying to bind a function (a callback) to a component.
The binding at first seems to work perfectly it the method is simple or not in an object.
The issue raised if the function bound is part of a typescript instance and it is calling some other method of that instance. The 'this' that is referering normally to the instance of the class is still refering to the instance of the 'controller'.

 this ends with a **"TypeError: this.internalMethodXXX() is not a function"**

**reproduce the bug**
I attached a archive containing some little adaptation from the aurelia-skeleton-typescript-webpack that show the issue.
[src.zip](https://github.com/aurelia/binding/files/413370/src.zip)

Basically :
* I create a 'UsersContainer' class that load the users and bind them to the users component. it also contains a method showUser that is called by the 'Users' class when a user is selected.
* the class Users is now just binding users and the callback. 

* **What is the expected behavior?**
I was expecting to be able to bind function of typescript class that are allowed to use their internal methods.

regards,

**Comment by doktordirk** [comment]
i think you need to use an arrow function for bound functions. like showUser = () => { .. }

**Comment by jphautin** [comment]
Hello,

And thanks for the reply. 

Following your comment, I am not sure what I have to modify, so I tried following syntax :
- show-user.bind="(user) => { showUser(user) }" ==> Error: Parser Error: Unexpected token > at column 9 in [(user) => { showUser(user) }]
- show-user="(user) => { showUser(user) }" ==> TypeError: this.showUser is not a function
- show-user="showUser" ==> failure this.showUser is not a function

Or perhaps you mean that aurelia binding does not allow binding a function of a class, even the controller class but only "standalone one". In this case, I think this should be explain in the documentation as I does not see how to use it.

regards,

regards,

**Comment by doktordirk** [comment]
the second (in you class) without the quotation marks

**Comment by jdanyow** [comment]
use the `.call` binding command to bind references to functions.

http://aurelia.io/hub.html#/doc/article/aurelia/binding/latest/binding-basics/4

**Comment by jphautin** [comment]
Sorry,

Still doesn't work because using call is not exactly what I need.

- show-user.call="showUser(user)"
I try this one first 'but in this case the 'user' object is lost during the call. the user is correct in the userSelected function but when calling the binding function it is set to undefined. So user call syntax is not working either this way (And I would say it is normal as the user is not set at this level).

- show-user.call="showUser"
the function is not mapped at all. no call to showUser method is done.

As I understood the documentation, call is used in case we want to pass object that are part of the template/controller. this is not what I am trying to do. 
I am trying to give a component a callback method that will be use in several places in the component itself (in a loop and a detailed subview). 

what I understand is that binding a function without knowing the parameters of it is not possible currently. Am I right ?

regards,
